universal-application-tool-0.0.1/app/services/program/predicate/OperatorRightHandType.java

```diff
@@ -0,0 +1,21 @@
+package services.program.predicate;
+
+/**
+ * An enum defining the types supported on the "right hand side" of an {@link Operator} in a {@link
+ * PredicateExpressionNode}, where the order of the expression looks like {@link
+ * services.question.types.ScalarType} {@link Operator} {@link OperatorRightHandType}.
+ *
+ * <ul>
+ *   For example:
+ *   <li>STRING IS_EQUAL_TO STRING
+ *   <li>DATE IS_AFTER DATE
+ *   <li>LONG NOT_IN LIST_OF_LONGS
+ * </ul>
+ */
+public enum OperatorRightHandType {
```

Comment:
question: now that we have this, does this clean up anything in the js for updating the value input type?

Reply:
That is a great question! I will check that out.

Reply:
It's not immediately obvious to me that this would help us clean up anything in the JS.